app/models/concerns/foreman_puppet/extensions/host.rb

```diff
@@ -61,6 +61,12 @@ def search_by_config_group(_key, operator, value)
       end
 
       module PrependedMethods
+        # TODO: we can drop this once extracted_from_core?
+        def validate_association_taxonomy(association_name)
+          return if association_name.to_sym == :environment
+          super
+        end
```

Comment:
```suggestion
        # TODO: we can drop this once extracted_from_core?
        unless ForemanPuppet.extracted_from_core?
          def validate_association_taxonomy(association_name)
            return if association_name.to_sym == :environment
            super
          end
        end
```

Reply:
we can do that, but I'd rather drop this method once we extract puppet from core.
If we accidently don't do that, this should be `noop`, so I feel the check is not necessary.

Reply:
if we add the `puppet_extracted` core branch to CI, won't there be a problem with tests without this check?

Reply:
Nope, this method will be noop

Reply:
there would be no such association on the `Host::Managed`